pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/kits/TestKitUtil.java

```diff
@@ -486,4 +512,48 @@ public String createKitRequestShipping(KitRequestShipping kitRequestShipping, DD
     public void deleteKitRequestShipping(int dsmKitRequestId) {
         kitDao.deleteKitRequestShipping(dsmKitRequestId);
     }
+
+    public void createEventsForDDPInstance(String eventName, String eventType, String eventDescription, boolean nullKitType) {
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult simpleResult = new SimpleResult();
+            try {
+                PreparedStatement stmt = conn.prepareStatement(INSERT_EVENT_TYPE, Statement.RETURN_GENERATED_KEYS);
+                stmt.setInt(1, ddpInstanceId);
+                stmt.setString(2, eventName);
+                stmt.setString(3, eventDescription);
+                if (nullKitType) {
+                    stmt.setNull(4, Types.INTEGER);
+                } else {
+                    stmt.setInt(4, kitTypeId);
+                }
+                stmt.setString(5, eventType);
+                stmt.executeUpdate();
+            } catch (SQLException e) {
```

Comment:
I think there's an antipattern here--the method returns void, but there's this two-step try/catch and then a check of results.resultException.  To reduce the complexity, instead of handling the return value from inTransaction this way, just do what you need to do within inTransaction, returning null, and then catch SQLException and retrhwo as DsmInternalError.